src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2516,18 +2480,23 @@ Status Session::receive_error_message(const ProtocolErrorInfo& info)
         return {ErrorCodes::SyncProtocolInvariantFailed, "Received ERROR message when it was not legal"};
     }
 
-    bool known_error_code = bool(get_protocol_error_message(info.raw_error_code));
-    if (REALM_UNLIKELY(!known_error_code)) {
-        return {ErrorCodes::SyncProtocolInvariantFailed,
-                util::format("Received ERROR message with unknown error code %1", info.raw_error_code)};
+    auto status = protocol_error_to_status(static_cast<ProtocolError>(info.raw_error_code), info.message);
+    // If the server has requested an action but sent an invalid error code we can still process the error and pass an
```

Comment:
I think the assumption is that the server always sends an action (cc @tkaye407), so maybe an assert is more appropriate.

Reply:
The server should always send an action, yes

Reply:
Addendum: we will only do it for realm devices that know about the JSONErrorMessage, but I dont think that matters in this case

Reply:
I'm going to fix this in a separate PR.